subprojects/core-api/src/main/java/org/gradle/api/artifacts/ResolvableDependencies.java

```diff
@@ -104,7 +104,7 @@ public interface ResolvableDependencies extends ArtifactView {
     void afterResolve(@DelegatesTo(ResolvableDependencies.class) Closure action);
 
     /**
-     * Returns the resolved dependency graph, performing the resolution if required. This will resolve the dependency graph but will not resolve or download the files.
+     * Returns the resolved dependency graph, performing the resolution if required when called on the resolution result. This will resolve the dependency graph but will not resolve or download the files.
```

Comment:
:x: What we need to convey is that resolution of the dependency graph will happen depending on which method is called on `ResolutionResult`

For example doing `this.resolutionResult.root` will cause resolution, because `root` was invoked on `ResolutionResult`.
But doing `this.resolutionResult.rootComponent` will not cause resolution, because a `Provider` is returned, which is a lazy value again. Resolution will only happen once `get()` (or other methods) is called on that result.

So I would propose to rephrase this as follows:

```
/**
 * Returns the resolved dependency graph, performing its resolution lazily.
 * <p/>
 * The lazy aspect depends on what is done with the returned {@code ResolutionResult}.
 * <p/>
 * This will resolve the dependency graph but will not resolve or download the artifacts.
```